cli/command/container/run.go

```diff
@@ -205,7 +209,7 @@ func runContainer(dockerCli command.Cli, opts *runOptions, copts *containerOptio
 	}
 
 	// Detached mode: wait for the id to be displayed and return.
-	if !config.AttachStdout && !config.AttachStderr {
+	if !attach {
```

Comment:
This is slightly different b/c `attach` is looking at AttachStdin as well. Is this intended?

Reply:
I'm not sure, my reading was that the original code (which didn't account for `AttachStdin`) was wrong -- because if you attach *just* stdin then `docker run` will exit prematurely (possibly not even accepting input) and won't wait for the container to exit.

For instance, `docker run -a stdin busybox tee /foo` will only read a single line of input before exiting. But if you want to keep the old behaviour, I can change this back.

Reply:
Hmm... I'm thinking if it's not material to fixing the issue you are seeing let's keep as is.

Reply:
I've removed it.